### DISCUSSION TITLE
Custom Types in help shows parser name

### DISCUSSION THREAD

**Discussion by afikbar** [fastapi/typer#618] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typing_extensions import Annotated


class CustomClass:
    def __init__(self, value: str):
        self.value = value

    def __str__(self):
        return f"<CustomClass: value={self.value}>"


def parse_custom_class(value: str):
    return CustomClass(value * 2)


def main(
    custom_arg: Annotated[CustomClass, typer.Argument(parser=parse_custom_class)],
    custom_opt: Annotated[CustomClass, typer.Option(parser=parse_custom_class)] = "Foo",
):
    print(f"custom_arg is {custom_arg}")
    print(f"--custom-opt is {custom_opt}")


if __name__ == "__main__":
    typer.run(main)
```

### Description

I've been trying to use typer with custom types, recently a version was released with such a support (thx!).

However, when specifying a `parser` for a `typer.Option(...)`, the "type" description in `--help` of the cli shows the parser function name as the argument type.

For example, using the example provided in [docs](https://typer.tiangolo.com/tutorial/parameter-types/custom-types):

```cmd
Usage: my-cli main [OPTIONS] CUSTOM_ARG

Arguments:
  CUSTOM_ARG  [required]

Options:
  --custom-opt PARSE_CUSTOM_CLASS
                                  [default: Foo]
  --help                          Show this message and exit.
```
(Note PARSE_CUSTOM_CLASS)

### Operating System

macOS

### Typer Version

0.9.0

### Python Version

3.9.13

**Comment by danmysak** [comment]
One way to handle this:

```python3
from typing import Callable


def name_function[T: Callable](function: T, name: str) -> T:
    """Set the name of a function."""
    function.__name__ = name
    return function


def named_function[T: Callable](name: str) -> Callable[[T], T]:
    """Decorator to set the name of a function."""
    return lambda function: name_function(function, name)


@named_function('custom type name to display or an empty string')
def parse_custom_class(value: str) -> CustomClass:
    return CustomClass(value * 2)
```
